runtime/compiler/control/JITServerCompilationThread.hpp

```diff
@@ -0,0 +1,117 @@
+using ClassOfStatic_t = UnorderedMap<std::pair<TR_OpaqueClassBlock *, int32_t>, TR_OpaqueClassBlock *>;
+using FieldOrStaticAttrTable_t = UnorderedMap<std::pair<TR_OpaqueClassBlock *, int32_t>, TR_J9MethodFieldAttributes>;
+
+namespace TR
+{
+// Objects of this type are instantiated at JITServer
+class CompilationInfoPerThreadRemote : public TR::CompilationInfoPerThread
+   {
+   public:
+   friend class TR::CompilationInfo;
+   CompilationInfoPerThreadRemote(TR::CompilationInfo &compInfo, J9JITConfig *jitConfig, int32_t id, bool isDiagnosticThread);
+   bool cacheIProfilerInfo(TR_OpaqueMethodBlock *method, uint32_t byteCodeIndex, TR_IPBytecodeHashTableEntry *entry);
+   TR_IPBytecodeHashTableEntry *getCachedIProfilerInfo(TR_OpaqueMethodBlock *method, uint32_t byteCodeIndex, bool *methodInfoPresent);
+
+   private:
+   /* Template method for allocating a cache of type T on the heap.
+    * Cache pointer must be NULL.
+    */
+   template <typename T>
+   bool initializePerCompilationCache(T* &cache)
+      {
+      // Initialize map
+      TR_ASSERT(!cache, "Cache already initialized");
+      TR_Memory *trMemory = getCompilation()->trMemory();
+      cache = new (trMemory->trHeapMemory()) T(typename T::allocator_type(trMemory->heapMemoryRegion()));
+      return cache != NULL;
+      }
+   /* Template method for storing key-value pairs (of types K and V respectively)
+    * to a heap-allocated unordered map.
+    * If a map is NULL, will allocate it.
+    */
+   template <typename K, typename V>
+   void cacheToPerCompilationMap(UnorderedMap<K, V>* &map, const K &key, const V &value)
+      {
+      if (!map)
+         initializePerCompilationCache(map);
+      map->insert({ key, value });
+      }
+
+   /* Template method for retrieving values from heap-allocated unordered map.
+    * If the map is NULL or value is not found, returns false.
+    * Otherwise, sets value to retrieved value and returns true
+    */
+   template <typename K, typename V>
+   bool getCachedValueFromPerCompilationMap(UnorderedMap<K, V>* &map, const K &key, V &value)
+      {
+      if (!map)
+         return false;
+      auto it = map->find(key);
+      if (it != map->end())
+         {
+         // Found entry at a given key
+         value = it->second;
+         return true;
+         }
+      return false;
+      }
+
+   /* Template method for clearing a heap-allocated cache.
+    * Simply sets pointer to cache to NULL.
+    */
+   template <typename T>
+   void clearPerCompilationCache(T* &cache)
+      {
+      // Since cache was heap-allocated,
+      // memory will be released automatically at the end of the compilation
+      cache = NULL;
+      }
+   TR_PersistentMethodInfo *_recompilationMethodInfo;
+   uint32_t _seqNo;
+   bool _waitToBeNotified; // accessed with clientSession->_sequencingMonitor in hand
+   IPTableHeap_t *_methodIPDataPerComp;
+#if defined(JITSERVER_TODO)
+   TR_ResolvedMethodInfoCache *_resolvedMethodInfoMap;
```

Comment:
Use
`class TR_ResolvedMethodInfoCache;`
and then you can eliminate the JITSERVER_TODO